core/trino-parser/src/main/antlr4/io/trino/sql/parser/SqlBase.g4

```diff
@@ -838,7 +838,17 @@ pathSpecification
     ;
 
 privilege
-    : CREATE | SELECT | DELETE | INSERT | UPDATE
+    : SELECT | UPDATE | INSERT | DELETE | CREATE
+    | LETTER+ (UNDERSCORE LETTER+)*
+    ;
+
+entityKind
+    : TABLE | SCHEMA
+    | LETTER+ (UNDERSCORE LETTER+)*
```

Comment:
This uses lexer tokens that are defined as "fragments" and are not visible to the parser. I suspect that as it currently exists, this production never matches. Do you have any tests that involve an entity kind that's not table or schema?

Reply:
I fixed the fragment creating the appropriate productions.

HOWEVER, when I built a test in `TestSqlParser` for exotic privileges and entity kinds, I immediately ran into an ambiguity between `GRANT role TO grantee` and `GRANT funky_privilege ON t TO grantee`.  Before my changes, the ambiguity doesn't  exist, because the privileges, `SELECT`, `CREATE`, etc, are keywords and therefore don't collide with role names.

I don't know how to fix this ambiguity.  Do you have any suggestions?

Reply:
I refactored the grammar and all tests now pass.